ui/src/app/core/settings.service.ts

```diff
@@ -0,0 +1,58 @@
+import {Injectable} from '@angular/core';
+import {AuthService} from '../core/auth.service';
+import {CapabilitiesService} from "./capabilities.service";
+import {DisplayField} from "../shared/model/DisplayField";
+
+type Settings = {
+  // Unset for basic auth users
+  user: string
+  // Unset for non-project-based job manager
+  project: string
+  v1: {
+    displayColumns: DisplayField[]
+  }
+};
+
+@Injectable()
+export class SettingsService {
+  currentSettings: Settings;
+
+  constructor(
+    private readonly authService: AuthService,
+    private readonly capabilitiesService: CapabilitiesService,
+    private readonly localStorage: Storage
+  ) {
+    this.currentSettings = JSON.parse(this.localStorage.getItem('settings'));
+    if (this.currentSettings === null) {
+      const capabilities = this.capabilitiesService.getCapabilitiesSynchronous();
+      let user: string;
+      if (this.authService.isAuthenticated()) {
+        user = this.authService.userId;
+      }
+      this.currentSettings = {
+        'user' : user,
+        'project' : null,
+        'v1': { 'displayColumns': capabilities.displayFields } };
```

Comment:
One thing to consider here is whether we actually want to store the default value. I think the answer is probably "no". If I've never modified my settings, and the server changes the capabilities displayFields, I should get whatever the latest default view is. Else we're somewhat arbitrarily binding the user to a view of the columns we happened to be displaying when they first logged in.

Reply:
How about, if `'displayColumns'` isn't set in their settings, going through each `displayField` from `capabilities` and setting it be be viewable as a column?

Reply:
Would need to see the code but that sounds like it would probably be fine. Ultimately I think as long as the following holds I'm good: store an array of `displayColumns` in LocalStorage IFF the user manually modified their display columns